### DISCUSSION TITLE
Import recipes from HTTP URL or Git repo

### DISCUSSION THREAD

**Discussion by eggplants** [cooklang/cooklang-app-android#2] [Ideas]
For users who have published Cooklang code in online Git repositories, it is convenient to be able to import from them.

My vault: https://github.com/eggplants/cooklang-recipes

**Comment by dubadub** [comment]
It's a valid ask and there's ton of repositories with recipes people created. I was thinking how to use GH as a backend.

But one thing I couldn't solve. Eventually (I hope very soon), I'm thinking to add recipe web-clipper and also QR code recipe sharing. In that case system will need write access to files and could be a problem with public read-only git repositories.

**Reply by FranGarc** [reply]
Importing would mean just "bring those recipes to my phone".  App as is right now it would mean I can use my phone to add your github recipes repositories and that would be sync'ed with _my sync folder_ on wherever I put it (my desktop in my case).

Import just _this recipe_ and _the whole repository_ could be an important distinction to make here, since most people would go "I just want this chocolate cake recipe", though I can see my less cook-friendly friends wanting to import my whole repo since they like how  I cook.


Having just one repository act as backend for everyone would be risky (and most likely messy). 
Having the user need to set up a git repository to use the app (bc, for example, you decide to ditch current sync'ing system) would mean making it more "technical" than the average user cares for.

**Comment by Quentintin** [comment]
Oh yeah, Own github repository (or multiple with self and shared repo?) as source of truth instead of cloud, or local synced folder, could be a very great idea ! On my side it's a green flag ✅.
